src/components/Table/components/Row.tsx

```diff
@@ -133,9 +133,9 @@ function RowImpl<R extends Kinded, S>(props: RowProps<R>): ReactElement {
 
           // Need to keep track of the expanded columns so we can add borders as expected for the header rows
           const numExpandedColumns = isExpanded
-            ? tableState.getExpandedColumns(maybeExpandedColumn)?.length
+            ? tableState.numberOfExpandedChildren(maybeExpandedColumn.id)
               ? // Subtract 1 if the column is hidden on expand, since we're not rendering it.
-                tableState.getExpandedColumns(maybeExpandedColumn).length - (maybeExpandedColumn.hideOnExpand ? 1 : 0)
+                tableState.numberOfExpandedChildren(maybeExpandedColumn.id) - (maybeExpandedColumn.hideOnExpand ? 1 : 0)
```

Comment:
Per the `hideOnExpand` - The amount of customization available in this table is a bit wild.

Reply:
Yeah :-) the "oh wait, the Nth column uses the N-1th column's rendering logic?..." really threw me for a loop. I didn't know about that functionality, but "get it"/makes sense...